.bazelrc

```diff
@@ -3,16 +3,34 @@ try-import submodules/dev-tools/.bazelrc
 # Specific Bazel build/test options.
 build --enable_platform_specific_config
 
-build:macos --cxxopt='-std=c++17'
-
-build:linux --cxxopt='-std=c++17'
-
-build:windows --cxxopt="/std:c++17"
+# Verbose failure logs when something goes wrong.
+build --verbose_failures
 
+# Use Clang as a default compiler on Windows.
 build:windows --compiler="clang-cl"
 
+# Compiler options.
+build:gcc --action_env=CC=gcc
 build:clang --action_env=CC=clang
 
+# By default, build eventuals in C++17 mode.
+build:linux --cxxopt=-std=c++17
+build:linux --host_cxxopt=-std=c++17
+build:macos --cxxopt=-std=c++17
+build:macos --host_cxxopt=-std=c++17
+build:windows --cxxopt=/std:c++17
+build:windows --host_cxxopt=/std:c++17
+
+# On Windows, `__cplusplus` is wrongly defined without this switch.
+# See https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/
+build:windows --copt=/Zc:__cplusplus
+
+# Make sure to include as little of windows.h as possible.
```

Comment:
Please elaborate: why? (compilation speed, correctness, other?)

(Whatever the reason, avoiding most of `windows.h` sounds great! "Why we did this" comments tend to be more useful than "What we did here" comments for maintainers looking at this code several years from now, though)

Reply:
Added a comment!